src/pkg/storage/filesystem.go

```diff
@@ -0,0 +1,104 @@
+package storage
+
+import (
+	"github.com/alcionai/clues"
+	"github.com/spf13/cast"
+
+	"github.com/alcionai/corso/src/internal/common/str"
+)
+
+const (
+	FilesystemPath = "path"
+)
+
+var fsConstToTomlKeyMap = map[string]string{
+	StorageProviderTypeKey: StorageProviderTypeKey,
+	FilesystemPath:         FilesystemPath,
+}
+
+type FilesystemConfig struct {
+	Path string
+}
+
+func buildFilesystemConfigFromMap(config map[string]string) (*FilesystemConfig, error) {
+	c := &FilesystemConfig{}
+
+	if len(config) > 0 {
+		c.Path = orEmptyString(config[FilesystemPath])
+	}
+
+	return c, c.validate()
+}
+
+func (c *FilesystemConfig) validate() error {
+	check := map[string]string{
+		FilesystemPath: c.Path,
+	}
+
+	for k, v := range check {
+		if len(v) == 0 {
+			return clues.Stack(errMissingRequired, clues.New(k))
+		}
+	}
+
+	return nil
+}
+
+func (c *FilesystemConfig) fsConfigsFromStore(kvg KVStoreGetter) {
+	c.Path = cast.ToString(kvg.Get(FilesystemPath))
+}
+
+// TODO(pandeyabs): Remove this. It's not adding any value.
+func fsOverrides(in map[string]string) map[string]string {
+	return map[string]string{
+		FilesystemPath: in[FilesystemPath],
+	}
+}
+
+var _ Configurer = &FilesystemConfig{}
+
+func (c *FilesystemConfig) ApplyConfigOverrides(
+	kvg KVStoreGetter,
+	readConfigFromStore bool,
+	matchFromConfig bool,
+	overrides map[string]string,
+) error {
+	if readConfigFromStore {
+		c.fsConfigsFromStore(kvg)
+
+		if matchFromConfig {
+			providerType := cast.ToString(kvg.Get(StorageProviderTypeKey))
+			if providerType != ProviderFilesystem.String() {
+				return clues.New("unsupported storage provider in config file: " + providerType)
+			}
+
+			// This is matching override values from config file.
+			if err := mustMatchConfig(kvg, fsConstToTomlKeyMap, fsOverrides(overrides)); err != nil {
+				return clues.Wrap(err, "verifying storage configs in corso config file")
+			}
+		}
+	}
+
+	c.Path = str.First(overrides[FilesystemPath], c.Path)
+
+	return c.validate()
+}
+
+// TODO(pandeyabs): We need to sanitize paths e.g. handle relative paths,
+// make paths cross platform compatible, etc.
```

Comment:
I think a call to `filepath.Abs(<input path>)` in the appropriate place will probably solve the issue. It should be done prior to writing values to the config file though (i.e. config file should contain absolute paths) otherwise things will get a little weird depending on the current working directory

I'm not sure what the symlink resolution is for `filepath.Abs`, but you can just leave a TODO for it probably

the `filepath` package is also already platform-specific, so it should handle all the checks for paths that don't adhere to the platform's expectations (i.e. wrong slash type) as well as give back the proper format

Reply:
Yeah! I wanted to address all of these in a later PR. I think it makes sense to have `filepath.Abs` atleast. Thanks!

Reply:
I've added support to make absolute paths & handle special cases. See `MakeAbsoluteFilePath`. It's currently being invoked by `filesystem` CLI code, which is fine since the `path` is sourced from `init` and `connect` CLI flags ( and eventually written to config file).

Ideally it should live in `FilesystemConfig` scope but that'd need some error propagation changes in `Configurer` interfaces. I'll handle that in a later PR, or as part of bigger config refactor.